src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -44,6 +44,14 @@ public class SocketTransportOptions
 
         public long? MaxWriteBufferSize { get; set; } = 64 * 1024;
 
+        /// <summary>
+        /// Inline application and transport continuations instead of dispatching to the threadpool.
+        /// </summary>
+        /// <remarks>
+        /// This will run application code on the IO thread which is why this is unsafe.
```

Comment:
Explain how this can hurt your performance. Also document the `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS` environment variable.